core/store/src/trie/mod.rs

```diff
@@ -698,7 +698,7 @@ impl Trie {
             storage,
             memtries,
             root,
-            charge_gas_for_trie_node_access: flat_storage_chunk_view.is_none(),
+            charge_gas_for_trie_node_access: false,
```

Comment:
Technically speaking this should be something more like the following. This one breaks replayability but I don't really mind to be honest given we actually only replay using tools and not this code path. 
`!ProtocolVersion::FlatStorage.enabled(protocol_version)`.